src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProviderSpec.scala

```diff
@@ -0,0 +1,125 @@
+package org.broadinstitute.dsde.workbench.leonardo.auth
+
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.typesafe.config.Config
+import org.broadinstitute.dsde.workbench.google.mock.MockGoogleIamDAO
+import org.broadinstitute.dsde.workbench.leonardo.CommonTestData
+import org.broadinstitute.dsde.workbench.leonardo.dao.MockGoogleDataprocDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.{DbSingleton, TestComponent}
+import org.broadinstitute.dsde.workbench.leonardo.model.NotebookClusterActions.{DeleteCluster, SyncDataToCluster}
+import org.broadinstitute.dsde.workbench.leonardo.model.ProjectActions.CreateClusters
+import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterName, ServiceAccountProvider}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.NoopActor
+import org.broadinstitute.dsde.workbench.leonardo.service.{LeonardoService, TestProxy}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+
+class TestSamAuthProvider(authConfig: Config, serviceAccountProvider: ServiceAccountProvider) extends SamAuthProvider(authConfig, serviceAccountProvider) {
+  override val samClient = new MockSwaggerSamClient()
+}
+
+class SamAuthProviderSpec extends FreeSpec with ScalatestRouteTest with Matchers with MockitoSugar with BeforeAndAfter with BeforeAndAfterAll with TestComponent with CommonTestData with TestProxy with ScalaFutures with OptionValues {
+  val googleProject = project.value
```

Comment:
None of these `val`s (`googleProject`, `clusterName`, `routeTest`) appear to be used